cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/deploy/pipeline/CreateSystemTablesStage.java

```diff
@@ -50,10 +50,14 @@ public void process(ApplicationDeployable input) throws IOException, TableAlread
 
         // If the table exists, check if the schema is compatible with the spec
         if (!schema.isCompatible(spec)) {
-          // don't allow deploying the app if the app expects a specification different than the one that exists
-          throw new IllegalArgumentException(
-            String.format("System table '%s' already exists, but with a different specification.",
-                          spec.getTableId().getName()));
+          if (spec.getNeedToAddFields()) {
```

Comment:
check below comment, you don't need the spec to tell you whether you need to add fields or not. Just let admin handle it.

Reply:
> @chtyim is there any reason we should get rid of the table registry? Understood it is not necessary, just curious on what benefit this can bring us.

I'm guessing that the postgresql table registry is still just another table that stores the registered tables from before if the tables are there already. It's not reading from DB directly and not most up to date?